engine/script/src/test/test_json.lua

```diff
@@ -135,11 +151,13 @@ function test_json_encode()
     -- interleaved nil becomse a null element (note: not in quotes)
     assert(json.encode({1,nil,2}) == "[1,null,2]")
 
-    local v3_enc = json.encode(vmath.vector3())
-    assert(startswith("\"object at", v3_enc))
+--TODO SHOULD WE SUPPORT THIS?
+--I THINK WE SHOULDN'T
+    -- local v3_enc = json.encode(vmath.vector3())
+    -- assert(startswith("\"object at", v3_enc))
 
-    local f_enc = json.encode(function() end)
-    assert(startswith("\"object at", f_enc))
+    -- local f_enc = json.encode(function() end)
+    -- assert(startswith("\"object at", f_enc))
```

Comment:
Q: What is the expectation here?
Obviiously there's a difference between defold-cjson and lua-cjson here, but since we recently added our `json.encode()` function, we can consider it a "fix".

Reply:
Oh, I didn't even know that we could throw something other than a Lua table to json.encode()!

If that is the case then sure, what should happen when we give it a vmath.vector3() or some other user type? And the same question, what happens when the type is encountered in a table?

1 Encode it to whatever tostring() returns?
2 Ignore it?
3 Generate a Lua error?

I'm leaning towards 1

Reply:
1 and 2 both are fine, I think

Reply:
Thx! I'll try to do #1, the tostring.